route/route.js

```diff
@@ -0,0 +1,205 @@
+            })
+        }
+    })
+})
+
+
+router.post('/recepient', (req, res) => {
+    if(!req.body.name || !req.body.email){
+        res.json({
+            success: false,
+            message: 'name and email required'
+        })
+    }else{
+        const newRecepient = new Recepient({
+            name: req.body.name,
+            email: req.body.email
+        })
+        newRecepient.save(err =>{
+            if(err){
+                res.json({
+                    success: false,
+                    message: 'error creating recepient'
+                })
+            }else{
+                res.json({
+                    success: true,
+                    message: 'recepient created'
+                })
+            }
+        })
+    }
+})
+router.post('/offer', (req, res) => {
+    if(!req.body.name || !req.body.discount){
+        res.json({
+            success: false,
+            message: 'name and discount required'
+        })
+    }else{
+        const newOffer = new Offer({
+            name: req.body.name,
+            discount: req.body.discount
+        })
+        newOffer.save(err =>{
+            if(err){
+                res.json({
+                    success: false,
+                    message: 'error creating offer'
+                })
+            }else{
+                res.json({
+                    success: true,
+                    message: 'offer created'
+                })
+            }
+        })
+    }
+})
+router.post('/voucher', (req, res) => {
+    if(!req.body.recepient || !req.body.offer){
+        res.json({
+            success: false,
+            message: 'recepient and offer required'
+        })
+    }else{
+        const code = uuid()
+        var tomorrow = new Date();
+        tomorrow.setDate(new Date().getDate()+1);
+        const newVoucher = new Voucher({  
+            recepient: req.body.recepient,
+            offer: req.body.offer,
+            code: code,
+            expiryDate: tomorrow
+        })
+        newVoucher.save(err =>{
+            if(err){
+                res.json({
+                    success: false,
+                    message: 'error creating offer'
+                })
+            }else{
+                res.json({
+                    success: true,
+                    message: 'voucher created',
+                    code: code
+                })
+            }
+        })
+    }
+})
+router.get('/verify-voucher',  (req, res) => {
+    if(!req.body.email || !req.body.offer){
+        res.json({
+            success: false,
+            message: 'recepient and offer required'
+        })
+    }else{             
+        Recepient.findOne({email : req.body.email}).then(recepient =>{
+            let id = recepient.id
+        Voucher.findOne({offer: req.body.offer, recepient: id}, (err, voucher)=>{
+            let exp = new Date(voucher.expiryDate)
+            let currDate = new Date()
+            if(voucher.used == false && currDate > exp ){
```

Comment:
```suggestion
            if(voucher.used == false && currDate < exp ){
```
since you want to know if the item has not expired, then the condition should be current time less than the expiry time.